src/martialarts.cpp

```diff
@@ -840,3 +843,171 @@ float ma_technique::armor_penetration( const player &u, damage_type type ) const
+        dump << _( "* Can eighter be used while <info>armed</info> or <info>unarmed</info>" ) << std::endl;
+    } else if( reqs.unarmed_allowed ) {
+        dump << _( "* Can <info>only</info> be used while <info>unarmed</info>" ) << std::endl;
+    } else if( reqs.melee_allowed ) {
+        dump << _( "* Can <info>only</info> be used while <info>armed</info>" ) << std::endl;
+    }
+
+    if( crit_tec ) {
+        dump << _( "* Will only activate on a <info>crit</info>" ) << std::endl;
+    }
+
+    if( dodge_counter ) {
+        dump << _( "* Will <info>counterattack</info> when you <info>dodge</info>" ) << std::endl;
+    }
+
+    if( block_counter ) {
+        dump << _( "* Will <info>counterattack</info> when you <info>block</info>" ) << std::endl;
+    }
+
+    if( miss_recovery ) {
+        dump << _( "* Will grant a <info>free recovery</info> from a <info>miss</info>" ) << std::endl;
+    }
+
+    if( grab_break ) {
+        dump << _( "* Will <good>break</good> a <info>grab</info>" ) << std::endl;
+    }
+
+    if( aoe == "wide" ) {
+        dump << _( "* Will attack in a <info>wide arc</info> in fron of you" ) << std::endl;
+
+    } else if( aoe == "spin" ) {
+        dump << _( "* Will attack <info>adjacent</info> enemies around you" ) << std::endl;
+
+    } else if( aoe == "impale" ) {
+        dump << _( "* Will <info>attack</info> your target and another <info>one behind</info> it" ) <<
+             std::endl;
+    }
+
+    if( knockback_dist ) {
+        dump << string_format( _( "* Will <info>knock back</info> enemies <stat>%d %s</stat>" ),
+                               knockback_dist, ngettext( "tile", "tiles", knockback_dist ) ) << std::endl;
+    }
+
+    if( down_dur ) {
+        dump << string_format( _( "* Will <info>down</info> enemies for <stat>%d %s</stat>" ),
+                               down_dur, ngettext( "turn", "turns", down_dur ) ) << std::endl;
+    }
+
+    if( stun_dur ) {
+        dump << string_format( _( "* Will <info>stun</info> target for <stat>%d %s</stat>" ),
+                               stun_dur, ngettext( "turn", "turns", stun_dur ) ) << std::endl;
+    }
+
+    if( disarms ) {
+        dump << _( "* Will <info>disarm</info> the target" ) << std::endl;
+    }
+
+    return dump.str().empty() ? description : dump.str();
+}
+
+bool ma_style_callback::key( const input_context &ctxt, const input_event &event, int entnum,
+                             uimenu *menu )
+{
+    const std::string action = ctxt.input_to_action( event );
+    if( action != "SHOW_DESCRIPTION" ) {
+        return false;
+    }
+    matype_id style_selected;
+    const size_t index = entnum;
+    if( index >= offset && index - offset < styles.size() ) {
+        style_selected = styles[index - offset];
+    }
+    if( !style_selected.str().empty() ) {
+        const martialart &ma = style_selected.obj();
+        std::ostringstream buffer;
+        if( !ma.techniques.empty() ) {
```

Comment:
Why are you checking this? The loop code will work fine even when `ma.techniques` is empty, the test is unnecessary.